db/event_manager_test.go

```diff
@@ -831,3 +831,126 @@ func assertChannelLengthWithTimeout(t *testing.T, c chan interface{}, expectedLe
+func TestWebhookHandleEventDBStateChangeFilterFuncError(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyEvents)()
+	ts, wr := InitWebhookTest()
+	defer ts.Close()
+	url := ts.URL
+	wr.Clear()
+	em := NewEventManager()
+	em.Start(2, 100)
+
+	warnCountPrev, err := strconv.Atoi(base.StatsResourceUtilization().Get(base.StatKeyWarnCount).String())
+	assert.NoError(t, err, "No error while getting the count of warnings")
+
+	source := `func (doc) { if (doc.state == "offline") { return true; } else { return false; } }`
+	timeout := uint64(30)
+	wh, _ := NewWebhook(fmt.Sprintf("%s/echo", url), source, &timeout)
+	em.RegisterEventHandler(wh, DBStateChange)
+	err = em.RaiseDBStateChangeEvent("db", "offline", "Index service is listening", "127.0.0.1:4985")
+	assert.NoError(t, err, "Error calling filter function")
+
+	func(warnCountPrev int) {
+		timeout := time.After(1000 * time.Millisecond)
+		tick := time.Tick(100 * time.Millisecond)
+		for {
+			select {
+			case <-timeout:
+				log.Fatalf("Retry timedout while reading request count and payload!")
+				return
+			case <-tick:
+				warnCountCurr, err := strconv.Atoi(base.StatsResourceUtilization().Get(base.StatKeyWarnCount).String())
+				assert.NoError(t, err, "No error while getting the count of warnings")
+				if warnCountCurr != warnCountPrev {
+					assert.Equal(t, 0, wr.GetCount(), "Request shouldn't be posted")
+					assert.Empty(t, wr.GetPayloads(), "Payload should be empty and request shouldn't be posted")
+					return
+				}
+			}
+		}
+	}(warnCountPrev)
+}
+
+// Simulate the scenario for handling unsupported events. Register an unsupported event manager; let'say EventType 255
+// and try to post database state change events; None of those events should be posted.
+func TestWebhookHandleUnsupportedEvenType(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyEvents)()
+	ts, wr := InitWebhookTest()
+	defer ts.Close()
+	url := ts.URL
+	wr.Clear()
+	em := NewEventManager()
+	em.Start(1, 100)
+
+	warnCountPrev, err := strconv.Atoi(base.StatsResourceUtilization().Get(base.StatKeyWarnCount).String())
+	assert.NoError(t, err, "No error while getting the count of warnings")
+	source := `function(doc) { if (doc.state == "online") { return true; } else { return false; } }`
+	timeout := uint64(30)
+	wh, _ := NewWebhook(fmt.Sprintf("%s/echo", url), source, &timeout)
+	// Event type 255 is not supported at the time of writing this test.
+	UnsupportedEvent := EventType(255)
+	em.RegisterEventHandler(wh, UnsupportedEvent)
+	err = em.RaiseDBStateChangeEvent("db", "online", "Index service is listening", "127.0.0.1:4985")
+	assert.NoError(t, err, "Database state change event should be raised successfully")
+
+	warnCountCurr, err := strconv.Atoi(base.StatsResourceUtilization().Get(base.StatKeyWarnCount).String())
+	assert.NoError(t, err, "No error while getting the count of warnings")
+	warnCount := warnCountCurr - warnCountPrev
+	assert.Equal(t, 0x1, warnCount)
```

Comment:
What's the reason for this being hex?

```suggestion
	assert.Equal(t, 1, warnCount)
```

Reply:
There is no need for this actually.